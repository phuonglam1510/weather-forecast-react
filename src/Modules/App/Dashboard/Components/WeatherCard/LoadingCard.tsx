import React from 'react';

import './WeatherCard.scss';

interface Props {}

const LoadingCard: React.FunctionComponent<Props> = () => {
  return (
    <div className="WeatherCard Loading">
      <div className="Title" />
      <div className="Text" />
    </div>
  );
};

export default LoadingCard;
