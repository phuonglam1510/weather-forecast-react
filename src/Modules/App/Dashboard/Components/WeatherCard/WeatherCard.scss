@import 'Styles/mixins';

.WeatherCard {
  background: #4583b6;
  border-radius: 6px;
  padding: 20px;
  color: white;
  h3 {
    color: #e2e47b;
    &.Today {
      font-weight: bold;
      font-size: 18px;
    }
  }
  .Name {
    font-weight: bold;
    font-style: italic;
  }

  img {
    width: 50px;
    object-fit: cover;
  }
  .Temp {
    span:first-of-type {
      font-weight: bold;
      font-size: 18px;
    }
    span {
      font-size: 18px;
    }
  }
  &.Loading {
    .Title {
      background-color: rgb(212, 212, 212);
      height: 30px;
    }
    .Text {
      background-color: rgb(236, 236, 236);
      margin-top: 10px;
      margin-right: 20px;
      height: 20px;
    }
  }
}
