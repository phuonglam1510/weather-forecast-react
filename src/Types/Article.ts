export interface IArticle {
  Id: number;
  Title: string;
  Content: string;
  PublicApproved: boolean;
  CoverImageURL: string;
}

interface ContentItem {
  id: string;
  text?: string;
  image?: any;
}

export class Article implements IArticle {
  Id: number;

  PerformerID: number;

  Title: string;

  Content: string;

  PublicApproved: boolean;

  CoverImageURL: string;

  constructor(json?: any) {
    this.Id = json.Id || '';
    this.PerformerID = json.PerformerID || '';
    this.Title = json.Title || '';
    this.Content = json.Content || '';
    this.PublicApproved = json.PublicApproved || false;
    this.CoverImageURL = json.CoverImageURL || '';
  }

  public get ContentItems(): ContentItem[] {
    try {
      const json = JSON.parse(this.Content);
      return json;
    } catch (error) {
      return [];
    }
  }

  public get Description(): string {
    return this.ContentItems.map(c => c.text || '').join(' ');
  }

  public get ThumbnailUrl(): string {
    return `${process.env.REACT_APP_API_URL}/api/image?id=${this.CoverImageURL}&quality=0`;
  }

  public get MainUrl(): string {
    return `${process.env.REACT_APP_API_URL}/api/image?id=${this.CoverImageURL}&quality=100`;
  }
}
