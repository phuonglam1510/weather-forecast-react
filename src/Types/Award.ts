export interface Award {
  ID: number;
  Name: string;
  Description: string;
  Year: number;
  ImageUrl: string;
  PerformerName: string;
}
