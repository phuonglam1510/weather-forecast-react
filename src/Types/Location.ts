import { LOCATION_TYPE } from 'Enums/Location';
import moment from 'moment';

export class Location {
  title: string;

  location_type: LOCATION_TYPE;

  woeid: number;

  coordinates: number[];

  timezone: string;

  time: Date;

  sun_rise: Date;

  sun_set: Date;

  timezone_name: string;

  parent?: Location;

  constructor(json?: any) {
    this.title = json.title || '';
    this.location_type = json.location_type || LOCATION_TYPE.CITY;
    this.woeid = json.woeid || -1;
    this.coordinates = json.latt_long ? json.latt_long.split(',') : [];
    this.timezone = json.timezone || '';
    this.time = new Date(json.time || '');
    this.sun_rise = new Date(json.sun_rise || '');
    this.sun_set = new Date(json.sun_set || '');
    this.timezone_name = json.timezone_name || '';
    if (json.parent) {
      this.parent = new Location(json.parent);
    }
  }

  get today() {
    return moment(this.time).format('DD-MM-YYYY');
  }

  get sunriseTime() {
    return moment(this.sun_rise).format('hh:mm a');
  }

  get sunsetTime() {
    return moment(this.sun_set).format('hh:mm a');
  }
}
