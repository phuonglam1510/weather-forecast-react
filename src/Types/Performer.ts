import moment from 'moment';

// types.ts
export interface Performer {
  id: string;
  firstName: string;
  lastName: string;
  stageName: string;
  nickName: string;
  gender: 'male' | 'female' | 'other';
  birthDate: Date;
  careerStartedYear: number;
  biography: string;
  status: string;
  minIncome: string;
  maxIncome: string;
  costNegotiable: boolean;
  loading: boolean;
  avatarURL: string;
}

export interface User {
  id: string;
  firstName: string;
  lastName: string;
  authEmail: string;
  dateOfBirth: Date;
  createdAt: Date;
  phone: string;
  loading: boolean;
  nickName?: string;
  avatarURL: string;
  status: 'active' | 'disabled';
  profile: {
    firstName: string;
    lastName: string;
    dateOfBirth: string;
    phone: string;
    avatarURL: string;
  };
}

export function fullName(person: Performer | User): string {
  if (!person) {
    return 'Not set';
  }
  if (person.nickName) {
    return person.nickName;
  }
  return `${person.firstName} ${person.lastName}`;
}

export function genderFormat(genderValue: string): string {
  return genderValue === 'male' ? 'Nam' : 'Ná»¯';
}

export function dateFormat(dateString: string | Date): string {
  return moment(dateString).format('DD-MM-YYYY');
}
