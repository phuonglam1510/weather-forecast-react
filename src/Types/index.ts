export * from './Location';
export * from './WeatherDay';
