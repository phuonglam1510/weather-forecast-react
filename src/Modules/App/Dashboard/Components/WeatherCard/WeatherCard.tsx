import React from 'react';

import './WeatherCard.scss';
import { WeatherDay } from 'Types';

interface Props {
  day: WeatherDay;
}

const WeatherCard: React.FunctionComponent<Props> = ({ day }) => {
  return (
    <div className="WeatherCard">
      <h3 className={day.isToday ? 'Today' : ''}>{day.dayName}</h3>
      <img src={day.icon} alt="day icon" />
      <div className="Name">{day.weather_state_name} </div>
      <div className="Temp">
        <span>{day.maxTemperature} </span>
        <span>{day.minTemperature}</span>
      </div>
    </div>
  );
};

export default WeatherCard;
