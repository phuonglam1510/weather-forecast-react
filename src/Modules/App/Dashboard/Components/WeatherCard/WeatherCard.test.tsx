/* eslint-disable import/no-extraneous-dependencies */
import React from 'react';
import renderer from 'react-test-renderer';
import { WeatherDay } from 'Types';
import WeatherCard from './WeatherCard';

test('Snapshot Test 1: Empty day', () => {
  const emptyDay = new WeatherDay({});
  const component = renderer.create(<WeatherCard day={emptyDay} />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Snapshot Test 2: Normal day', () => {
  const emptyDay = new WeatherDay({
    air_pressure: 1018.5,
    applicable_date: '2020-05-06',
    created: '2020-05-07T03:16:18.059367Z',
    humidity: 45,
    id: 5140782216380416,
    max_temp: 21.715,
    min_temp: 12.7,
    predictability: 70,
    the_temp: 24.125,
    visibility: 16.632864073808953,
    weather_state_abbr: 'lc',
    weather_state_name: 'Light Cloud',
    wind_direction: 288.16761556308336,
    wind_direction_compass: 'WNW',
    wind_speed: 5.50213065348233,
  });
  const component = renderer.create(<WeatherCard day={emptyDay} />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
