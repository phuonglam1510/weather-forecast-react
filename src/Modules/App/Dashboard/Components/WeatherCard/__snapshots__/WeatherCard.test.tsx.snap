// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot Test 1: Empty day 1`] = `
<div
  className="WeatherCard"
>
  <h3
    className=""
  >
    Invalid date
  </h3>
  <img
    alt="day icon"
    src="undefined/static/img/weather/.svg"
  />
  <div
    className="Name"
  >
    -1
     
  </div>
  <div
    className="Temp"
  >
    <span>
      0째
       
    </span>
    <span>
      0째
    </span>
  </div>
</div>
`;

exports[`Snapshot Test 2: Normal day 1`] = `
<div
  className="WeatherCard"
>
  <h3
    className=""
  >
    Yesterday
  </h3>
  <img
    alt="day icon"
    src="undefined/static/img/weather/lc.svg"
  />
  <div
    className="Name"
  >
    Light Cloud
     
  </div>
  <div
    className="Temp"
  >
    <span>
      22째
       
    </span>
    <span>
      13째
    </span>
  </div>
</div>
`;
